server/src/machines/dre/api.rs

```diff
@@ -0,0 +1,102 @@
+use std::time::Duration;
+use crate::serial::devices::dre::DreData;
+use super::DreMachine;
+use control_core::{
+    machines::api::MachineApi,
+    socketio::{
+        event::{Event, GenericEvent},
+        namespace::{
+            CacheFn, CacheableEvents, Namespace, NamespaceCacheingLogic, NamespaceInterface,
+            cache_duration,
+        },
+    },
+};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use uom::si::f64::Length;
+
+
+#[derive(Serialize, Debug, Clone)]
+pub struct DiameterEvent {
+    pub dre_data: Option<DreData>,
+}
```

Comment:
PLease don't send `DreData`, just send the `diameter: f64` and write a comment that it's a `/// Diameter in mm`